patroni/ctl.py

```diff
@@ -1318,19 +1330,15 @@ def _do_failover_or_switchover(obj: Dict[str, Any], action: str, cluster_name: s
 @ctl.command('failover', help='Failover to a replica')
 @arg_cluster_name
 @option_citus_group
-@click.option('--leader', '--primary', '--master', 'leader', help='The name of the current leader', default=None)
```

Comment:
This is a non-backwards compatible breaking change, so I wonder if we should instead leave the option in but print a warning that it will be removed in a future version. Then remove it when we bump the major version of Patroni to 4. Perhaps we can add a warning message like this?

```
if leader:
  logger.warning('Supplying a leader name using this command is deprecated and will be removed in a future version of Patroni, change your scripts to use `switchover` instead.')
```

Reply:
yes, I can do that but I am not sure if we need 4 for this. is 3.2 not enough?
And to be honest, just the option removal is not the most breaking change here... We also have "allow manual failover to an async node in sync mode" and it basically will not be possible to use `patronictl failover` as a replacement for unscheduled `patronictl switchover` anymore. So I wonder if the warning with the leader value discarding will actually hide the extent of the changes... 
@CyberDem0n wdyt

Reply:
Actually, I like the idea about showing a warning. Not sure whether we should discard the provided leader value or not.
Waiting for 4 shouldn't be a problem. There is another "breaking" change pending - replacing "master" with primary everywhere.

Reply:
> Not sure whether we should discard the provided leader value or not.

if we don't - the whole PR doesn't make much sense

Reply:
I agree adding the warning is a good idea but it should more descriptive about "it is not be possible to use patronictl failover as a replacement for unscheduled patronictl switchover anymore" Basically, big part of the `patronictl failover` behaviour has changed

Reply:
> > Not sure whether we should discard the provided leader value or not.
> 
> if we don't - the whole PR doesn't make much sense

I think that it's sometimes good to introduce a bunch of code to support a new way of doing things which introduces a breaking change, which will do all the work but defaults to not making that change at first. Either the code is not called or is available as an "opt-in" configuration. A warning shows that said code will become the new standard and allow people time to switch. Then finally it becomes active by switching the config option to be the default or just deactivating the old way but not removing the code. If the old way is still needed then it can be activated by a user. Then in a future version that feature can be removed when the user has had time to migrate.

You might not need all that song-and-dance for this, but I don't think the change is wasted.

Reply:
> yes, I can do that but I am not sure if we need 4 for this. is 3.2 not enough?

That would not be using semantic versioning. Major version change is needed when the code is no longer compatible with a user's use of it. I say wait for 4 as you might want to queue up a few changes to do together so we're not on v143 by Christmas ;-)

Reply:
Ok, giving up, you've convinced me:) 
so something like this? https://github.com/zalando/patroni/pull/2784/commits/cc336fc1fe08d1514fb2026faf516b2c87a0a779

Reply:
Looks good! Added a comment.

Reply:
extended the docstring